Services/UI/OJS.Services.Ui.Business/Implementations/ContestCategoriesBusinessService.cs

```diff
@@ -71,17 +81,23 @@ public async Task<IEnumerable<ContestCategoryListViewModel>> GetAllParentCategor
         return categories;
     }
 
-    private void AddChildren(
-        IEnumerable<ContestCategoryTreeViewModel> children,
-        ICollection<ContestCategoryTreeViewModel> allCategories)
-        => children.ForEach(child =>
+    private void AddChildren(IEnumerable<ContestCategoryTreeViewModel> children,
+        ICollection<ContestCategoryTreeViewModel> allCategories,
+        ICollection<AllowedContestStrategiesServiceModel> parentCategoryAllowedStrategyTypes) =>
+        children.ForEach(child =>
         {
+            child.AllowedStrategyTypes =
+                this.contestCategoriesData.GetAllAllowedStrategyTypes<AllowedContestStrategiesServiceModel>(child.Id)
```

Comment:
This is **N+1 query**. You are calling `                this.contestCategoriesData.GetAllAllowedStrategyTypes<AllowedContestStrategiesServiceModel>(child.Id)` for each child of each category.

These must be fetched beforehand in memory with a single call - i.e.
```
var allowedStrategyTypes = GetAllowedStrategyTypesForCategories(categoryIds)
```